tokio/src/runtime/handle.rs

```diff
@@ -28,21 +32,131 @@ pub(crate) struct Handle {
     pub(super) blocking_spawner: blocking::Spawner,
 }
 
+#[derive(Debug)]
+pub struct HandleEnterGuard(EnterGuard);
```

Comment:
We shouldn't need to define a new struct. It should be possible to modify `runtime::EnterGuard` to be backed by `Handle`. `Runtime::enter` can then delegate to `Handle::enter`.

Reply:
I wasn't convinced by this new type either. Your proposal sounds way better indeed!
Testing a fix